### PR TITLE
Rwa 900 update pact specification to v4

### PR DESCRIPTION
**Pull request by serdaroguzhanakin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-900

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskBySearchForCompletablePactTest.java [hunk 1]
```diff
@@ -86,10 +86,21 @@ public void getTasksBySearchForCompletableCriteriaWithWarnings() {
...
+                PermissionTypes.MANAGE,
+                PermissionTypes.REFER
+            )
+        );
+
         Task task = new Task(
             "4d4b6fgh-c91f-433f-92ac-e456ae34f72a",
             "Review the appeal",
-            "reviewTheAppeal",
+            "ReviewTheAppeal",
```
Comment (mkayad) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskBySearchForCompletablePactTest.java:103]: Should not task type (ReviewTheAppeal) start with a lower case as before?

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskBySearchForCompletablePactTest.java [hunk 2]
```diff
@@ -107,36 +118,37 @@ public void getTasksBySearchForCompletableCriteriaWithWarnings() {
...
+                PermissionTypes.MANAGE,
+                PermissionTypes.REFER
+            )
+        );
+
         Task taskWithWarnings = new Task(
             "4d4b6fgh-c91f-433f-92ac-e456ae34f72a",
             "Review the appeal",
-            "reviewTheAppeal",
+            "ReviewTheAppeal",
```
Comment (mkayad) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskBySearchForCompletablePactTest.java:151]: Same comment as before

And also this FT should not be delete:
https://github.com/hmcts/wa-task-management-api/blob/52f5167b776df841773cb6a9f5fa1ee9ac1ddd04/src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java

**Review by alexbaeza — DISMISSED** [review]

File: build.gradle [hunk 3]
```diff
@@ -347,31 +350,31 @@ dependencies {
...
-  contractTestCompile group: 'au.com.dius.pact.provider', name: 'junit5', version: versions.pactVersion
-  contractTestCompile group: 'au.com.dius.pact.provider', name: 'spring', version: versions.pactVersion
-  contractTestCompile group: 'au.com.dius.pact.provider', name: 'junit5spring', version: versions.pactVersion
+  contractTestImplementation group: 'au.com.dius.pact.provider', name: 'junit5', version: versions.pactVersion
+  contractTestImplementation group: 'au.com.dius.pact.provider', name: 'spring', version: versions.pactVersion
+  contractTestImplementation group: 'au.com.dius.pact.provider', name: 'junit5spring', version: versions.pactVersion
 
-  contractTestCompile('org.junit.jupiter:junit-jupiter-api:5.7.2')
-  contractTestRuntime('org.junit.jupiter:junit-jupiter-engine:5.7.2')
+//  contractTestCompile('org.junit.jupiter:junit-jupiter-api:5.7.2')
```
Comment (BenNzewi) [inline build.gradle:376]: is this commented out for a reason?

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/ccd/util/PactDslBuilderForCaseDetailsList.java [hunk 4]
```diff
@@ -17,7 +17,7 @@ private PactDslBuilderForCaseDetailsList() {
     public static DslPart buildStartEventForCaseWorkerPactDsl(String eventId) {
         return newJsonBody(o ->
             o.stringType("event_id", eventId)
-                .stringType("token", null)
+                .stringType("token", "")
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/ccd/util/PactDslBuilderForCaseDetailsList.java:20]: will a token ever be an empty string? Better to pass a "dummyToken"

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/ccd/util/PactDslBuilderForCaseDetailsList.java [hunk 5]
```diff
@@ -31,7 +31,7 @@ public static DslPart buildStartEventForCaseWorkerPactDsl(String eventId) {
     public static DslPart buildStartForCaseWorkerPactDsl(String eventId) {
         return newJsonBody(
             o -> o.stringType("event_id", eventId)
-                .stringType("token", null)
+                .stringType("token", "")
                 .object("case_details", cd -> {
                     cd.stringMatcher("jurisdiction", ALPHABETIC_REGEX, "IA");
                     cd.stringMatcher("case_type_id", ALPHABETIC_REGEX, "Asylum");
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/ccd/util/PactDslBuilderForCaseDetailsList.java:37]: will a token ever be an empty string? Better to pass a "dummyToken"

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskProviderTest.java [hunk 6]
```diff
@@ -126,6 +124,17 @@ private void setInitMockTaskWithWarnings() {
     }
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagementGetTaskProviderTest.java:124]: the following seem are deprecated in version 4.3.4 ->
 scheme = "${PACT_BROKER_SCHEME:http}",
    host = "${PACT_BROKER_URL:localhost}",
    port = "${PACT_BROKER_PORT:9292}",

Can we try using the "url" as suggested in the doc on the pr for one providers and see if it works on our pact broker?

**Review by mkayad — APPROVED** [review]
LGTM

**Review by mkayad — APPROVED** [review]
LGTM

**Pull request merged**